raster-foundry/raster-foundry
# Add Platform Email Settings Storage Backend

## Overview

This PR adds backend support for storing platform email information and notification preferences. It also updates the associated property tests.

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

## Testing Instructions

 * Run migration
 * Set dev user as RF platform admin
 * Sign in as dev user and grab a token
 * `GET` to `api/platforms/<platform id>` should include `publicSettings` containing platform email user name, smtp host, ingest notification setting, aoi notification setting (no email password)
 * `PUT` to `api/platforms/<platform id>` with json look something like below:
```
{
    "id": "<platform ID>",
    "name": "<platform name>",
    "publicSettings": {
        "emailUser": "<email user string>",
        "emailSmtpHost": "<host string>",
        "emailIngestNotification": <boolean>,
        "emailAoiNotification": <boolean>
    },
    "isActive": <boolean>,
    "defaultOrganizationId": "<org ID>",
    "privateSettings": {
    	"emailPassword": "<password string>"
    }
}
```
 * `GET` again should see the updated info (no email password)

Closes #3443
